### PR TITLE
Fix PHPStan for Connectivity context

### PR DESCRIPTION
**Pull request by pchasle**
Add PHPStan analysis for Connectivity context to the CI

Fix static analysis issues:
- Level 8 on _Domain_ & _Application_
- Level 5 on _Infrastructure_

Add a missing Integration test for `DbalWrongCredentialsCombinationRepositoryIntegration.php`

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Repository/DbalWrongCredentialsCombinationRepositoryIntegration.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        sleep(1);
+
+        $this->repository->create(new Write\WrongCredentialsCombination('erp', 'admin'));
+        $current = $this->dbalConnection->executeQuery($sql)->fetchAll();
+
+        Assert::assertNotSame($previous, $current);
+    }
+
+    public function test_it_finds_all_the_wrong_credentials_conbination(): void
```
Comment (pierrejolly) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Repository/DbalWrongCredentialsCombinationRepositoryIntegration.php:94]:
```suggestion
    public function test_it_finds_all_the_wrong_credentials_combination(): void
```

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCount.php [hunk 2]
```diff
@@ -25,6 +25,9 @@ public function __construct(string $connectionCode, int $count)
         $this->count = $count;
     }
 
+    /**
+     * @return array{connection_code: string, count: int}
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCount.php:29]: what ? Where does this format come from?

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCountPerConnection.php [hunk 3]
```diff
@@ -28,6 +28,9 @@ public function __construct(array $errorCounts)
         }
     }
 
+    /**
+     * @return array<string, int>
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCountPerConnection.php:32]: Where does this format come from?

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionCode.php [hunk 4]
```diff
@@ -11,24 +11,24 @@
...
+    /** @var string */
     private $code;
 
     public function __construct(string $code)
     {
         $code = trim($code);
 
         if (empty($code)) {
-            throw new \InvalidArgumentException(sprintf(self::CONSTRAINT_KEY, 'required'));
+            throw new \InvalidArgumentException('akeneo_connectivity.connection.connection.constraint.code.required');
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionCode.php:22]: Why ?

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Cli/UpdateAuditDataCommand.php [hunk 5]
```diff
@@ -70,7 +70,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     private function purgeOlderThanXDays(int $days): void
     {
         $before = new \DateTimeImmutable("now - $days days", new \DateTimeZone('UTC'));
-        $before->setTime((int) $before->format('H'), 0, 0);
+        $before = $before->setTime((int) $before->format('H'), 0, 0);
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Cli/UpdateAuditDataCommand.php:73]: Good catch. Did we have a bug on it?

File: src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCount.php (hunk 2 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Domain/Audit/Model/Read/ErrorCount.php:29]: My bad, I should have added to the description of the PR.
It's the phpdoc format of phpstan, that allow the static analyzer to interpret arrays and check for missing key, wrong property, value type...

See https://phpstan.org/writing-php-code/phpdoc-types#array-shapes

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionCode.php (hunk 4 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/ValueObject/ConnectionCode.php:22]: To enhance code discovery, be able to retrieve a constraint more easily in the code.
There is no gain in splitting the constraint on a constant here.
(but yes, nothing to do with phpstan)

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Cli/UpdateAuditDataCommand.php (hunk 5 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Cli/UpdateAuditDataCommand.php:73]: Good catch from phpstan :)
I guess we had small bug, but what was not purged would have been the day after, so nothing critical.

**Pull request merged**